fgrpc/grpcrunner.go

```diff
@@ -41,16 +41,28 @@ const (
 )
 
 // Dial dials grpc either using insecure or using default tls setup.
-// TODO: option to specify certs.
-func Dial(serverAddr string, tls bool) (conn *grpc.ClientConn, err error) {
-	opts := grpc.WithInsecure()
-	if tls || (strings.HasPrefix(serverAddr, "https://")) {
-		opts = grpc.WithTransportCredentials(credentials.NewTLS(nil))
+func Dial(serverAddr string, cert string) (conn *grpc.ClientConn, err error) {
+	var opts []grpc.DialOption
+	//var creds credentials.TransportCredentials
+	switch {
+	case cert != "":
+		creds, err := credentials.NewClientTLSFromFile(cert, "")
```

Comment:
I updated the name of this PR to properly reflect the implemented functionality. This PR originally provided mTLS, but I simplified things based on your initial feedback. This PR will allow clients (i.e. grpcping) to authenticate the Fortio gRPC server and encrypt the data exchanged. In my original PR, the server cert was fetched from the CA (Fortio server). Since refactoring, the server cert must be shipped with the client-side application. [This](https://bbengfort.github.io/programmer/2017/03/03/secure-grpc.html) blog provides details of the original and current approach for this PR. Let me know how you would like to proceed.

Reply:
[This section](https://en.wikipedia.org/wiki/Transport_Layer_Security#Basic_TLS_handshake) provides a detailed explanation of the basic TLS handshake implemented by this PR.